reckon-core/src/main/groovy/org/ajoberstar/reckon/core/GitVcs.java

```diff
@@ -0,0 +1,239 @@
+ */
+package org.ajoberstar.reckon.core;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.zafarkhaja.semver.ParseException;
+import com.github.zafarkhaja.semver.Version;
+import org.eclipse.jgit.lib.*;
+import org.eclipse.jgit.errors.*;
+import org.eclipse.jgit.revwalk.*;
+import org.eclipse.jgit.revwalk.filter.*;
+
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import java.util.Comparator;
+import java.util.Optional;
+import java.util.Objects;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.Map;
+import java.util.List;
+import java.util.HashSet;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+public class GitVcs implements Vcs {
+  private static final Logger logger = LoggerFactory.getLogger(GitVcs.class);
+
+  private final Repository repo;
+  private final Function<Ref, Optional<Version>> tagParser;
+
+  public GitVcs(Repository repo) {
+    this(
+        repo,
+        ref -> {
+          try {
+            String tagName = Repository.shortenRefName(ref.getName());
+            return Optional.of(Version.valueOf(tagName));
+          } catch (ParseException e) {
+            return Optional.empty();
+          }
+        });
+  }
+
+  public GitVcs(Repository repo, Function<Ref, Optional<Version>> tagParser) {
+    this.repo = repo;
+    this.tagParser = tagParser;
+  }
+
+  @Override
+  public Optional<Version> getCurrentVersion() {
+    return Optional.of(getBaseVersion()).filter(BaseVersion::isAnyHead).map(BaseVersion::getAny);
+  }
+
+  @Override
+  public Optional<Version> getPreviousRelease() {
+    return Optional.of(getBaseVersion().getNormal());
+  }
+
+  @Override
+  public Optional<Version> getPreviousVersion() {
+    return Optional.of(getBaseVersion().getAny());
+  }
+
+  private BaseVersion getBaseVersion() {
+    try (RevWalk walk = new RevWalk(repo)) {
+      walk.setRetainBody(false);
+
+      ObjectId headObjectId = repo.getRefDatabase().getRef("HEAD").getObjectId();
+      RevCommit headCommit = walk.parseCommit(headObjectId);
+
+      Set<GitVersion> versions = new HashSet<>();
+
+      for (Ref ref : repo.getRefDatabase().getRefs(Constants.R_TAGS).values()) {
+        Ref tag = repo.peel(ref);
+        ObjectId objectId = tag.getPeeledObjectId();
+        RevCommit revCommit = walk.parseCommit(objectId);
+        Optional<GitVersion> maybeVersion =
+            tagParser.apply(tag).map(version -> new GitVersion(version, revCommit));
+
+        if (maybeVersion.isPresent()) {
+          GitVersion version = maybeVersion.get();
+          // walk.reset();
+          // walk.setRevFilter(RevFilter.MERGE_BASE);
+          // walk.markStart(version.getRevCommit());
+          // walk.markStart(headCommit);
+          //
+          // RevCommit mergeBase = walk.next();
+
+          versions.add(version);
+          // if (!headCommit.equals(mergeBase)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.ajoberstar.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)